### PR TITLE
T1 51.auth back refactor

### PR DESCRIPTION
**Pull request by diegoplatap**
## Global user authentication 

- Using the auth middleware now users are authenticated to access private routes.
- Created a loadUser controller that let the frontend have the user data even if the page is refreshed.

Example: the route (/me) in routes.js includes the auth middleware and return some user data. 
image

### CODE REVIEW

File: controllers/controllers.js [hunk 1]
```diff
@@ -1,6 +1,34 @@
+const User = require("../models/User");
+const Foundation = require("../models/Foundation");
+var jwt = require("jsonwebtoken");
+const config = require("../config/index");
 Pet = require("../models/Pet");
```
Comment (leoneloliveros) [inline controllers/controllers.js:5]: you need to add type of declaration (const, let, var)

File: routers/routes.js [hunk 2]
```diff
@@ -1,11 +1,12 @@
 const express = require("express");
-
 const controllers = require("../controllers/controllers");
-
+const { auth } = require("../middlewares/middlewares");
 const app = express.Router();
 
+app.get("/me", auth, controllers.loadUser);
+app.post("/login", controllers.login);
 app.delete("/pets/:id", controllers.destroyPet);
```
Comment (leoneloliveros) [inline routers/routes.js:8]: Add middleware to not public routes

File: controllers/controllers.js (hunk 1 above)
Comment (diegoplatap) [reply controllers/controllers.js]: Add type of declaration in controllers variables

File: routers/routes.js (hunk 2 above)
Comment (diegoplatap) [reply routers/routes.js]: Added the "auth" middleware in not public routes

**Review by leoneloliveros — APPROVED** [review]

**Pull request merged**